### PR TITLE
Feature/35384

### PR DESCRIPTION
**Pull request by szafran89**
Issue https://github.com/SnowdogApps/magento2-alpaca-components/issues/67

### CODE REVIEW

File: components/02-elements/mega-menu/_mega-menu.scss [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+        }
+
+        &--inner {
+            margin: 0;
+            padding: 0;
+            font-size: $mega-menu__font-size;
+            font-weight: normal;
+            &:focus,
+            &:hover {
+                color: $color-primary;
```
Comment (anqaka) [inline components/02-elements/mega-menu/_mega-menu.scss]: shouldn't it be moved to the component variable?

File: components/02-elements/mega-menu/_mega-menu.scss [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+            }
+        }
+
+        &--inner-heading {
+            margin: 0 0 $spacer--medium;
```
Comment (anqaka) [inline components/02-elements/mega-menu/_mega-menu.scss]: this spacer shouldn't be moved to the component variable?

File: components/02-elements/mega-menu/_mega-menu.scss [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+        padding: $mega-menu__dropdown-padding;
+        background-color: $mega-menu__dropdown-background-color;
+        box-shadow: $mega-menu__dropdown-box-shadow;
+        position: absolute;
+        top: 100%;
+        left: 0;
+        list-style: none;
+        z-index: 2;
+        @include mq($screen-l) {
+            max-width: 850px;
```
Comment (anqaka) [inline components/02-elements/mega-menu/_mega-menu.scss]: add to component variable?

File: components/02-elements/mega-menu/mega-menu.config.js [hunk 4]
```diff
@@ -0,0 +1,153 @@
+module.exports = {
+  title: 'Mega menu',
+  collated: true,
```
Comment (anqaka) [inline components/02-elements/mega-menu/mega-menu.config.js]: we don't have variants, so it's no necessary

File: components/02-elements/mega-menu/mega-menu.hbs [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+                                       mega-menu__link--inner-heading
+                                   "
+                                >
+                                    {{ heading }}
+                                </a>
+
+                                {{> '@list--link'
+                                    listTag="ul"
+                                    elementTag="li"
+                                }}
```
Comment (anqaka) [inline components/02-elements/mega-menu/mega-menu.hbs]: I am wondering if it shouldn't be a module if we use another element inside it? What do you think?

File: components/02-elements/mega-menu/mega-menu.hbs [hunk 6]
```diff
@@ -0,0 +1,55 @@
+<div class="mega-menu">
+    <ul class="mega-menu__wrapper container">
+        {{#each list}}
+            <li class="mega-menu__item {{ class }}">
+                <a href="#" class="mega-menu__link">
```
Comment (anqaka) [inline components/02-elements/mega-menu/mega-menu.hbs]: this `href` value should be set in config? B/c this item should have some link, not be an empty link?

File: components/02-elements/mega-menu/mega-menu.hbs [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+                        {{#each nestedList}}
+                            <li class="
+                                    mega-menu__dropdown-item
+                                    mega-menu__dropdown-item--level1
+                                    {{#if modifier}}
+                                        {{modifier}}
+                                    {{/if}}
+                                "
+                            >
+                                <a href="#"
```
Comment (anqaka) [inline components/02-elements/mega-menu/mega-menu.hbs]: same as above - IMO this `href` value should be set in config

File: components/02-elements/mega-menu/mega-menu.hbs [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+
+                        <li class="
+                                mega-menu__dropdown-item
+                                mega-menu__dropdown-item--image
+                            "
+                        >
+                            {{> '@image'
+                                src="/images/banner/banner-480_480.png"
+                                additionalClasses="mega-menu__image"
+                            }}
```
Comment (anqaka) [inline components/02-elements/mega-menu/mega-menu.hbs]: another element used here... so I'm afraid it should be a module...

File: components/02-elements/side-menu/_side-menu.scss [hunk 9]
```diff
@@ -1,28 +1,30 @@
...
+$side-menu__transform--open : translateX(100%) !default;
 $side-menu__link-color      : $gray-dark !default;
 $side-menu__link-color-hover: $gray !default;
 
 .side-menu {
     position: absolute;
     top: $side-menu__top;
-    left: $side-menu__left;
-    width: $side-menu__width;
+    right: $side-menu__left;
```
Comment (anqaka) [inline components/02-elements/side-menu/_side-menu.scss]: this variable shouldn't be named `$side-menu__right`?

File: components/02-elements/side-menu/side-menu.hbs [hunk 10]
```diff
@@ -1,10 +1,3 @@
 <div class="side-menu {{ class }}">
-    {{> '@list' list }}
-
-    <a href="#" class="side-menu__link link">
-        Example
-    </a>
-    <a href="#" class="side-menu__link link">
-        Sign in
-    </a>
+    {{ render '@dropdown-list--with-nested' }}
```
Comment (anqaka) [inline components/02-elements/side-menu/side-menu.hbs]: shouldn't it be a module if we use another element inside this one?

File: components/02-elements/mega-menu/_mega-menu.scss [hunk 11]
```diff
@@ -0,0 +1,103 @@
...
+$mega-menu__font-size--large          : 16px !default;
+$mega-menu__text-decoration           : none !default;
+$mega-menu__background-color          : $white !default;
+$mega-menu__container-background-color: $white !default;
+$mega-menu__justify-content           : center !default;
+$mega-menu__link-padding              : 12px $spacer--medium !default;
+$mega-menu__link-margin               : 0 $spacer--medium !default;
+
+$mega-menu__dropdown-padding          : 0 0 0 112px !default;
+$mega-menu__dropdown-box-shadow       : 0 4px 6px 0 rgba(51, 51, 51, 0.3) !default;
```
[the remark this answers is not in the document]
Comment (szafran89) [reply components/02-elements/mega-menu/_mega-menu.scss]: It isn't. Unfortunately I can't see `box-shadow` defined inside `variables.scss`

File: components/02-elements/mega-menu/_mega-menu.scss (hunk 1 above)
Comment (szafran89) [reply components/02-elements/mega-menu/_mega-menu.scss]: moved.

File: components/02-elements/mega-menu/_mega-menu.scss (hunk 2 above)
Comment (szafran89) [reply components/02-elements/mega-menu/_mega-menu.scss]: moved.

File: components/02-elements/mega-menu/mega-menu.config.js (hunk 4 above)
Comment (szafran89) [reply components/02-elements/mega-menu/mega-menu.config.js]: done.

File: components/02-elements/mega-menu/_mega-menu.scss (hunk 3 above)
Comment (szafran89) [reply components/02-elements/mega-menu/_mega-menu.scss]: added.

File: components/02-elements/mega-menu/mega-menu.hbs (hunk 6 above)
Comment (szafran89) [reply components/02-elements/mega-menu/mega-menu.hbs]: added to config.

File: components/02-elements/mega-menu/mega-menu.hbs (hunk 7 above)
Comment (szafran89) [reply components/02-elements/mega-menu/mega-menu.hbs]: changed.

File: components/02-elements/mega-menu/mega-menu.hbs (hunk 5 above)
Comment (szafran89) [reply components/02-elements/mega-menu/mega-menu.hbs]: moved to modules

File: components/02-elements/mega-menu/mega-menu.hbs (hunk 8 above)
Comment (szafran89) [reply components/02-elements/mega-menu/mega-menu.hbs]: 👍

File: components/02-elements/side-menu/side-menu.hbs (hunk 10 above)
Comment (szafran89) [reply components/02-elements/side-menu/side-menu.hbs]: moved.

File: components/02-elements/mega-menu/_mega-menu.scss (hunk 11 above)
[the remark this answers is not in the document]
Comment (anqaka) [reply components/02-elements/mega-menu/_mega-menu.scss]: ok, so let's leave it like that :)

File: components/03-modules/mega-menu/mega-menu.config.js [hunk 12]
```diff
@@ -0,0 +1,177 @@
...
+          {
+            class: '',
+            heading: 'Example1',
+            link: '/example1',
+            elements: [
+              {
+                class: '',
+                text: 'Lorem ipsum',
+                link: '/lorem-ipsum',
+                elementLink: '#',
```
Comment (anqaka) [inline components/03-modules/mega-menu/mega-menu.config.js]: we have `link` and `elementLink` I cannot find usage of `elementLink` in while project. could you please check it and remove if not used?

File: components/03-modules/side-menu/side-menu.config.js [hunk 13]
```diff
@@ -0,0 +1,67 @@
...
+  context: {
+    list: [
+      {
+        text: 'Furniture',
+        class: 'mega-menu__item mega-menu__item--level0 mega-menu__item--parent',
+        nestedList: [
+          {
+            class: 'mega-menu__inner-item mega-menu__inner-item--level2',
+            text: 'Lorem ipsum',
+            elementLink: '#',
```
Comment (anqaka) [inline components/03-modules/side-menu/side-menu.config.js]: `elementLink` it's not used, to add `href="#"` you should use `itemAttributes` from the `dropdown-list` config, please check every element with `a href` here

Comment (szafran89) [reply components/03-modules/side-menu/side-menu.config.js]: it's legacy code, I think we can remove whole file. Is it ok?

File: components/03-modules/mega-menu/mega-menu.config.js (hunk 12 above)
Comment (szafran89) [reply components/03-modules/mega-menu/mega-menu.config.js]: Yes, you're right. I've removed it.

File: components/03-modules/side-menu/side-menu.config.js (hunk 13 above)
Comment (anqaka) [reply components/03-modules/side-menu/side-menu.config.js]: Indeed, we don't need the config file for this... I thought that the config is needed to display the component 👯  but as we see it's working well, so IMO it's ok :)

**Comment by anqaka** [comment]
code is ok

**Pull request merged**